packages/yarnpkg-core/sources/structUtils.ts

```diff
@@ -565,10 +577,14 @@ export function parseFileStyleRange(range: string, {protocol}: {protocol: string
   return {parentLocator, path};
 }
 
+const ENCODE_25 = /%/g;
+const ENCODE_3A = /:/g;
+const ENCODE_23 = /#/g;
+
 function encodeUnsafeCharacters(str: string) {
-  str = str.replace(/%/g, `%25`);
-  str = str.replace(/:/g, `%3A`);
-  str = str.replace(/#/g, `%23`);
+  str = str.replace(ENCODE_25, `%25`);
+  str = str.replace(ENCODE_3A, `%3A`);
+  str = str.replace(ENCODE_23, `%23`);
```

Comment:
```suggestion
  str = str.replaceAll(`%`, `%25`);
  str = str.replaceAll(`:`, `%3A`);
  str = str.replaceAll(`#`, `%23`);
```
https://node.green/#ES2021-features--String-prototype-replaceAll
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replaceAll#browser_compatibility